superset-frontend/src/dashboard/components/DeleteComponentButton.tsx

```diff
@@ -16,18 +16,15 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { MouseEventHandler } from 'react';
 import Icons from 'src/components/Icons';
 import IconButton from './IconButton';
 
-const propTypes = {
-  onDelete: PropTypes.func.isRequired,
+type DeleteComponentButtonProps = {
+  onDelete: MouseEventHandler<HTMLDivElement>;
 };
 
-const defaultProps = {};
-
-export default class DeleteComponentButton extends React.PureComponent {
+class DeleteComponentButton extends React.PureComponent<DeleteComponentButtonProps> {
   render() {
     const { onDelete } = this.props;
```

Comment:
> Can you add the default value for `onChange` please? I'm not sure if it's actually needed, but it won't hurt to keep it

_Originally posted by @kgabryje in https://github.com/apache/superset/pull/18120#discussion_r789613146_

Can you add the default value for `onDelete` please?

Reply:
Sure thing!
I've added the default value.